ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/RaoDataManager.java

```diff
@@ -128,13 +130,15 @@ public void fillCracResultsWithSensis(double cost, double overCost) {
         updateCnecExtensions();
     }
 
-    public void fillCracResultsWithLoopFlowConstraints(Map<String, Double> loopFlows, Map<Cnec, Double> loopFlowShifts) {
+    public void fillCracResultsWithLoopFlowConstraints(Map<String, Double> loopFlows, Map<Cnec, Double> loopFlowShifts, Network network) {
         raoData.getCrac().getCnecs(raoData.getCrac().getPreventiveState()).forEach(cnec -> {
             CnecLoopFlowExtension cnecLoopFlowExtension = cnec.getExtension(CnecLoopFlowExtension.class);
+
             if (!Objects.isNull(cnecLoopFlowExtension)) {
+                double loopFlowThreshold = Math.abs(cnecLoopFlowExtension.getInputThreshold(Unit.MEGAWATT, network));
                 double initialLoopFlow = Math.abs(loopFlows.get(cnec.getId()));
-                double loopFlowThreshold = Math.abs(cnecLoopFlowExtension.getInputLoopFlow());
-                cnecLoopFlowExtension.setLoopFlowConstraint(Math.max(initialLoopFlow, loopFlowThreshold));
+
+                cnecLoopFlowExtension.setLoopFlowConstraintInMW(Math.max(initialLoopFlow, loopFlowThreshold - cnec.getFrm()));
```

Comment:
Once again I think the FRM management is unclear here

Reply:
What we want for the loop flow constraint is : 
(1 - minRamFactor) * Fmax - frm

where Fmax is the flow limit on the Cnec (as defined in the CBCORA file, without considering the frm)

The problem is that in our crac-api, the getMaxThreshold() method return (Fmax - frm), and not Fmax.

So basically, we are doing : 
(1 - minRamFactor) * (getMaxThreshold + FRM) - FRM

This is indeed unclear, but the crac-api as it is now is quite unclear also, and I refrain myself from refactoring anything :yum:

Reply:
Ok I get it ! And where is the (1 - minRamFactor) ?

Reply:
sorry, (1 - minRamFactor) would be (inputThreshold) defined with Unit.PERCENT_IMAX.

the (1-minRamFactor) computation is handled in the CBCORA importer

Reply:
Alright ! All clear !